sv_pipeline/load_data.py

```diff
@@ -0,0 +1,392 @@
+#!/usr/bin/env python
+
+import argparse
+from datetime import datetime
+from elasticsearch import helpers as es_helpers
+import re
+import subprocess
+from tqdm import tqdm
+
+from hail_scripts.shared.elasticsearch_client import ElasticsearchClient
+from hail_scripts.shared.elasticsearch_utils import ELASTICSEARCH_INDEX
+
+GS_SAMPLE_PATH = 'gs://seqr-datasets/v02/GRCh38/RDG_WES_Broad_Internal/base/projects/{project_guid}/{project_guid}_{file_ext}'
+
+GENCODE_CHR_COL_IDX = 0
+GENCODE_TYPE_COL_IDX = 2
+GENCODE_START_COL_IDX = 3
+GENCODE_END_COL_IDX = 4
+GENCODE_INFO_COL_IDX = 8
+
+CHR_COL = 'chr'
+START_COL = 'start'
+END_COL = 'end'
+QS_COL = 'QS'
+CN_COL = 'CN'
+CALL_COL = 'svtype'
+SAMPLE_COL = 'sample'
+NUM_EXON_COL = 'genes_any_overlap_totalExons'
+DEFRAGGED_COL = 'defragmented'
+SC_COL = 'vac'
+SF_COL = 'vaf'
+VAR_NAME_COL = 'name'
+GENES_COL = 'genes_any_overlap_Ensemble_ID'
+IN_SILICO_COL = 'path'
+
+CHROM_FIELD = 'contig'
+SAMPLE_ID_FIELD = 'sample_id'
+GENOTYPES_FIELD = 'genotypes'
+CN_FIELD = 'cn'
+QS_FIELD = 'qs'
+GENES_FIELD = 'geneIds'
+TRANSCRIPTS_FIELD = 'sortedTranscriptConsequences'
+SF_FIELD = 'sf'
+SC_FIELD = 'sc'
+VARIANT_ID_FIELD = 'variantId'
+CALL_FIELD = 'svType'
+DEFRAGGED_FIELD = 'defragged'
+NUM_EXON_FIELD = 'num_exon'
+
+BOOL_MAP = {'TRUE': True, 'FALSE': False}
+
+COL_CONFIGS = {
+    CHR_COL: {'field_name': CHROM_FIELD, 'format': lambda val: val.lstrip('chr')},
+    SC_COL: {'field_name': SC_FIELD, 'format': int},
+    SF_COL: {'field_name': SF_FIELD, 'format': float},
+    VAR_NAME_COL: {'field_name': VARIANT_ID_FIELD, 'format': lambda val, call='any': '{}_{}'.format(val, call)},
+    CALL_COL: {'field_name': CALL_FIELD},
+    START_COL: {'format': int},
+    END_COL: {'format': int},
+    QS_COL: {'field_name': QS_FIELD, 'format': int},
+    CN_COL: {'field_name': CN_FIELD, 'format': int},
+    NUM_EXON_COL: {'field_name': NUM_EXON_FIELD, 'format': lambda val: 0 if val == 'NA' else int(val)},
+    DEFRAGGED_COL: {'field_name': DEFRAGGED_FIELD, 'format': lambda val: BOOL_MAP[val]},
+    IN_SILICO_COL: {
+        'field_name': 'StrVCTVRE_score',
+        'format': lambda val: None if val == 'not_exonic' else float(val),
+        'allow_missing': True,
+    },
+    SAMPLE_COL: {
+        'field_name': SAMPLE_ID_FIELD,
+        'format': lambda val: re.search('(\d+)_(?P<sample_id>.+)_v\d_Exome_GCP', val).group('sample_id'),
+    },
+    GENES_COL: {'field_name': GENES_FIELD, 'format': lambda genes: [gene.split('.')[0] for gene in genes.split(',')]},
+}
+
+CORE_COLUMNS = [CHR_COL, SC_COL, SF_COL, CALL_COL, GENES_COL]
+SAMPLE_COLUMNS = [START_COL, END_COL, QS_COL, CN_COL,  NUM_EXON_COL, DEFRAGGED_COL, SAMPLE_COL]
+COLUMNS = CORE_COLUMNS + SAMPLE_COLUMNS + [VAR_NAME_COL]
+
+IN_SILICO_COLS = [VAR_NAME_COL, CALL_COL, IN_SILICO_COL]
+
+QS_BIN_SIZE = 10
+
+CHROMOSOMES = [
```

Comment:
this is a basic question, do we get chrM SV calls? (it looks like there aren't any in the latest callset)

Reply:
I copied this utility over from SNV utilities, so quite possibly no. Do you think there is harm in leaving this in?

Reply:
probably not, was just curious